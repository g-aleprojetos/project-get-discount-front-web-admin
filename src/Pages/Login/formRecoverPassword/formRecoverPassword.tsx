import React, {useCallback, useRef} from 'react';
import {UseFormHandleSubmit, UseFormRegister, useForm} from 'react-hook-form';
import {z} from 'zod';
import {zodResolver} from '@hookform/resolvers/zod';
import {Input} from 'Components/input';
import {Popup} from 'Components/popup';
import colors from 'resources/colors';
import * as S from './formRecoverPassword.styles';

const emailSchema = z.object({
  email: z.string().email(),
});

type EmailSchema = z.infer<typeof emailSchema>;

type Props = {
  handleEsqueceuSenha: () => void;
};

export const FormRecoverPassword = ({handleEsqueceuSenha}: Props) => {
  const {register, handleSubmit} = useForm<EmailSchema>({
    resolver: zodResolver(emailSchema),
  });

  const formRef = useRef<HTMLFormElement>(null);

  const handleSubmitEmail = useCallback((data: EmailSchema) => {
    console.log(data);
  }, []);

  return (
    <RecuperaPassword
      handleEsqueceuSenha={handleEsqueceuSenha}
      register={register}
      handleSubmit={handleSubmit}
      handleSubmitEmail={handleSubmitEmail}
      formRef={formRef}
    />
  );
};

interface RecuperaPasswordProps {
  handleEsqueceuSenha: () => void;
  register: UseFormRegister<EmailSchema>;
  handleSubmit: UseFormHandleSubmit<EmailSchema>;
  handleSubmitEmail: (data: EmailSchema) => void;
  formRef: React.RefObject<HTMLFormElement>;
}

const RecuperaPassword = ({
  handleEsqueceuSenha,
  register,
  handleSubmit,
  handleSubmitEmail,
  formRef,
}: RecuperaPasswordProps) => {
  return (
    <Popup
      textTitle="Recuperar senha"
      aoSair={handleEsqueceuSenha}
      botaoConfirmar={{
        legenda: 'Enviar',
        corTexto: colors.white,
        corBotao: colors.caribbeanGreen,
        aoClicar: () => {
          if (formRef.current) {
            formRef.current.dispatchEvent(
              new Event('submit', {cancelable: true, bubbles: true}),
            );
          }
        },
      }}
      botaoCancelar={{
        legenda: 'Cancelar',
        corTexto: colors.white,
      }}>
      <S.Form ref={formRef} onSubmit={handleSubmit(handleSubmitEmail)}>
        <Input
          testId="input-recuperar-senha"
          placeholder="contato@email.com"
          autoComplete="on"
          tipo="email"
          {...register('email')}
        />
      </S.Form>
    </Popup>
  );
};
