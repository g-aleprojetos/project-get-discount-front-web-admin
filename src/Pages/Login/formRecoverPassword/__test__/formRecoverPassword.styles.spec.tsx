import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../formRecoverPassword.styles';

describe('formRecoverPassword.styles', () => {
  test('Form DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<S.Form />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
