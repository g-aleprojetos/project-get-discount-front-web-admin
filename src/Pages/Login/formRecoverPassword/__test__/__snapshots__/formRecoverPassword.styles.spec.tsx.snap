// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formRecoverPassword.styles Form DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

<form
  className="c0"
/>
`;
